File: frontend/src/types/account.ts
export interface AccountRead {
  id: number;
  user_id: number;
  name: string;
  broker: string | null;
  currency: string;
  created_at: string;
}

export interface AccountCreateInput {
  name: string;
  broker?: string | null;
  currency?: string;
}

File: backend/app/schemas/account.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AccountCreate(BaseModel):
    name: str
    broker: str | None = None
    currency: str = "USD"


class AccountRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    name: str
    broker: str | None
    currency: str
    created_at: datetime
